tests/test_freshdesk_start_date.py

```diff
@@ -0,0 +1,166 @@
+import os
+
+from tap_tester import connections, runner
+
+from base import FreshdeskBaseTest
+
+
+class FreshdeskStartDateTest(FreshdeskBaseTest):
+
+    start_date_1 = ""
+    start_date_2 = ""
+    test_streams = {}
+
+    @staticmethod
+    def name():
+        return "tap_tester_freshdesk_start_date_test"
+
+    def get_properties(self, original: bool = True):
+        """Configuration properties required for the tap."""
+        return_value = {
+            #'start_date' : '2015-02-03T00:00:00Z',
+            'start_date' : '2015-02-06T00:00:00Z',
+        }
+        if original:
+            return return_value
+
+        return_value["start_date"] = self.start_date
+        return return_value
+
+    def test_run(self):
+        """Instantiate start date according to the desired data set and run the test"""
+
+        self.start_date_1 = self.get_properties().get('start_date')
+        self.start_date_2 = self.timedelta_formatted(self.start_date_1, days=7*365)
+
+        self.start_date = self.start_date_1
+
+        # Excluding broken streams: time_settings, satisfaction_ratings
+        # TODO spike on the two 403 streams above
+        test_streams = {'agents', 'companies', 'groups', 'tickets', 'conversations', 'roles'}
+        self.test_streams = test_streams
+        obey_start_date_streams = {'agents', 'companies', 'groups', 'roles', 'tickets', 'conversations'}
+
+        ##########################################################################
+        ### First Sync
+        ##########################################################################
+
+        # instantiate connection
+        conn_id_1 = connections.ensure_connection(self)
+
+        # run check mode
+        check_job_name_1 = self.run_and_verify_check_mode(conn_id_1)
+
+        # run initial sync
+        record_count_by_stream_1 = self.run_and_verify_sync(conn_id_1)
+        synced_records_1 = runner.get_records_from_target_output()
+
+        # Update based on sync data
+        first_sync_empty = self.test_streams - synced_records_1.keys()
+        if len(first_sync_empty) > 0:
+            print("Missing stream: {} in sync 1. Removing from test_streams. Add test data?".format(first_sync_empty))
+        first_sync_bonus = synced_records_1.keys() - self.test_streams
+        if len(first_sync_bonus) > 0:
+            print("Found stream: {} in first sync. Add to test_streams?".format(first_sync_bonus))
```

Comment:
Same as bookmarks, would be best to fail at end if we lose coverage on a stream.

Reply:
Updated.